### PR TITLE
Add sample-app CI testing

### PR DESCRIPTION
**Pull request by Dasomeone**
# What is this PR?

This PR introduces a number of significant changes to this repository.

Primarily, the basic structure has been altered to be split into:
- `.github/` for workflow files
- `ops/` for workflow supporting files and configurations, e.g. docker files and scripts
- `charts/` for helm charts used in deploying sample-apps on Kubernetes
- `sample-apps/` containing the sample-apps themselves.

Additionally, a new matrix workflow has been added for detecting changes to sample-apps, running on PRs against Main:
1. Check if sample apps have been changed
2. for each sample app:
    1. Check if the sample app is CI compatible
    2. Install Multipass & configure it
    3. Run Loki and Mimir within Multipass using Docker (fully isolated from other matrix jobs)
    4. Dynamically get the Loki & Mimir IP
    6. Spin up sample app, using `make LOKI_INSTANCE=${{ env.LOKI }} PROMETHEUS_INSTANCE=${{ env.MIMIR }} run-ci` which expects the sample app to make the configuration required, apply it, start the sample-app and start any load-generation required in the background
    7. Wait for the sample-app to generate metrics. Set using `SAMPLE_APP_SETUP_TIME=` in `.config`. (Default `180s` as set in global `.defaultconfig`)
    8. Query Mimir for each of the metrics contained within the sample-apps `expected_metrics` file and PASS/FAIL sample-app as required. (Required success rate defaults to `0.8` as set in `.defaultconfig` but can be overridden as required per sample-app)

Added support for sample apps to bypass CI by having `.CI_BYPASS` files in their sample app directories

Current sample-apps supporting CI execution:
- OpenLDAP

Current sample-apps with CI bypass set:
- apache-solr
- catchpoint
- istio
- openstack
- pgbouncer
- velero
- windows-active-directory

### CODE REVIEW

**Comment by Dasomeone** [comment]
Note: the previous Sample apps have not been tested, but are assumed still working.
Have a WIP conversion ongoing for pgBouncer to follow up soon as well

**Comment by Dasomeone** [comment]
Please note re: the warning comments, the bypassed sample apps have not been re-validated as part of this as they've not been changed in any way.

Known issue here is the comments can end up being somewhat spammy, so I'm looking into a way to update/check if the comment has already been placed, but that's a backlog task for the future for now.
If folks disagree with that, we can easily disable the warning.

File: sample-apps/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+1. Ensure the sample-app is built using Multipass
+   - Sample should be stand-alone and not contain any databases
+1. Ensure sample-app has load-generation scripts or capabilities to accurately produce metrics for the majority of use-cases
+1. Ensure sample-app name matches integration slug
+1. Add an `expected_metrics` file to the sample-app dir, containing a newline separated list of Prometheus metrics expected by the integration, e.g:
+    ```
+    up
+    metric_name_1
+    metric_name_2
```
Comment (v-zhuravlev) [inline sample-apps/README.md:23]: Could you metrics with expected labels? i.e. metric_nam_1{label1="value"}

Comment (Dasomeone) [reply sample-apps/README.md:23]: Not at this time no, though that can easily be added as a separate test suite in a second step!
The intent here is not so much to check labels, or anything specifically with regards to them, but more to check if all the metrics expected by the dashboards in the integration are present.

The `expected_metrics` file is equivalent to the `metric_names` files found within our cloud integrations in their repository, and an upcoming PR will automatically update them if matching sample apps are found.

As such, it's only required to be added manually when first adding a new sample app.

Also, in this case, I've included `metric_name_1` and `metric_name_2` manually to test that it still passes with a >= 80% match rate. They will be removed in a subsequent PR

**Review by v-zhuravlev — APPROVED** [review]
LGTM

**Pull request merged**